internal/services/network/network_manager_management_group_connection_resource.go

```diff
@@ -0,0 +1,245 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	managementParse "github.com/hashicorp/terraform-provider-azurerm/internal/services/managementgroup/parse"
+	managementValidate "github.com/hashicorp/terraform-provider-azurerm/internal/services/managementgroup/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/network/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/network/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+	"github.com/tombuildsstuff/kermit/sdk/network/2022-05-01/network"
+)
+
+type ManagerManagementGroupConnectionModel struct {
+	Name              string `tfschema:"name"`
+	ManagementGroupId string `tfschema:"management_group_id"`
+	ConnectionState   string `tfschema:"connection_state"`
+	Description       string `tfschema:"description"`
+	NetworkManagerId  string `tfschema:"network_manager_id"`
+}
+
+type ManagerManagementGroupConnectionResource struct{}
+
+var _ sdk.ResourceWithUpdate = ManagerManagementGroupConnectionResource{}
+
+func (r ManagerManagementGroupConnectionResource) ResourceType() string {
+	return "azurerm_network_manager_management_group_connection"
+}
+
+func (r ManagerManagementGroupConnectionResource) ModelObject() interface{} {
+	return &ManagerManagementGroupConnectionModel{}
+}
+
+func (r ManagerManagementGroupConnectionResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.NetworkManagerManagementGroupConnectionID
+}
+
+func (r ManagerManagementGroupConnectionResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"management_group_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: managementValidate.ManagementGroupID,
+		},
+
+		"network_manager_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
```

Comment:
shouldn't this be ForceNew?

Reply:
The resource ID is like `/providers/Microsoft.Management/managementGroups/%s/providers/Microsoft.Network/networkManagerConnections/%s` , the `network_manager_id` is in the [property](https://github.com/Azure/azure-rest-api-specs/blob/df77748f7dff3fe12c13b7262e9b307c719e6918/specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/networkManagerConnection.json#L371-L389), which is not a `ForceNew`.